File: printjson.py
import sys

EMAIL_FROM='andre@delphix.com'
TEMPLATE_NAME='usersgroupmodelo1'
CONFIG_SET='usersgroup'
DEFAULT_NAME='Caro'

if len(sys.argv) < 2:
    print("Lack of parameter")
    quit()
cont=-1
size=0
sizelen=0

try:
   arqin = open(sys.argv[1], 'r')
   data = arqin.readlines()
except OSError as err:
    print("OS error: {0}".format(err))
except IOError as err:
   print("IO error: {0}".format(err))
else:
   arqin.close()
   print("{\n  \"Source\": \"", EMAIL_FROM, "\",\n  \"Template\": \"",TEMPLATE_NAME,"\",\n  \"ConfigurationSetName\": \"",CONFIG_SET,"\",\n  \"Destinations\":[", sep='', end='\n', file=sys.stdout)
   sizelen=len(data)
   for eachLine in data:
      cont=cont+1
      valores=eachLine.split(' ')
      email = valores[0]
      nome = valores[1].rstrip('\n')
      print("    {\n      \"Destination\":{\n       \"ToAddresses\":[\n       \"", email,"\"\n       ]\n       },\n       \"ReplacementTemplateData\":\"{ \\\"name\\\":\\\"", nome, "\\\"}\"\n       }", sep='', end='\n', file=sys.stdout)
      size=size+1
	  
      if(cont>=49):
         print("  ],\n  \"DefaultTemplateData\":\"{ \\\"name\\\":\\\"", DEFAULT_NAME, "\\\"}\"\n}\n\n", sep='', end='\n', file=sys.stdout)
         print("{\n  \"Source\": \"", EMAIL_FROM, "\",\n  \"Template\": \"",TEMPLATE_NAME,"\",\n  \"ConfigurationSetName\": \"",CONFIG_SET,"\",\n  \"Destinations\":[", sep='', end='\n', file=sys.stdout)
         cont=-1;
      elif(cont>=0 and cont<49 and size!=sizelen):
         print(",", cont, sep='', end='\n', file=sys.stdout)

if(cont>=0 and cont<50):
    print("  ],\n  \"DefaultTemplateData\":\"{ \\\"name\\\":\\\"", DEFAULT_NAME, "\\\"}\"\n}\n\n", sep='', end='\n', file=sys.stdout)

quit()

File: printjson.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#define EMAIL_FROM "andre@delphix.com"
#define TEMPLATE_NAME "usersgroupmodelo1"
#define CONFIG_SET "usersgroup"
#define DEFAULT_NAME "Caro"

int main(int argc, char *argv[]) {
   FILE *arqin;
   char linha[160];
   char *nome, *email, *str;
   int cont=0;
      
   memset(linha, '\0', sizeof(linha));

   arqin = fopen(argv[1], "r+");
   if(arqin==NULL) {
      fprintf(stdout,"Erro ao abrir arquivo: %s\n",argv[1]);
      exit(1);
   }

   fseek(arqin, 0, SEEK_SET);
   cont=-1;
   fprintf(stdout, "{\n  \"Source\": \"%s\",\n  \"Template\": \"%s\",\n  \"ConfigurationSetName\": \"%s\",\n  \"Destinations\":[\n", EMAIL_FROM, TEMPLATE_NAME, CONFIG_SET);

   while( fgets(linha, sizeof(linha), arqin) != NULL) {
      cont++;
      if(cont>0 && cont<49) {
         fprintf(stdout,",\n");
      } else {
         if(cont>=48) {
            fprintf(stdout, "  ],\n  \"DefaultTemplateData\":\"{ \\\"name\\\":\\\"%s\\\"}\"\n}\n\n\n", DEFAULT_NAME);
            fprintf(stdout, "{\n  \"Source\": \"%s\",\n  \"Template\": \"%s\",\n  \"ConfigurationSetName\": \"%s\",\n  \"Destinations\":[\n", EMAIL_FROM, TEMPLATE_NAME, CONFIG_SET);
            cont=-1;
         }
      }
      str=(char *)malloc(sizeof(linha));
      linha[strlen(linha)-1]='\0';
      strcpy(str, linha);
      email = strsep(&str, " ");
      nome = strsep(&str, " ");
      fprintf(stdout, "    {\n      \"Destination\":{\n       \"ToAddresses\":[\n       \"%s\"\n       ]\n       },\n       \"ReplacementTemplateData\":\"{ \\\"name\\\":\\\"%s\\\"}\"\n       }\n", email, nome);
      nome=NULL;
      email=NULL;
      memset(linha, '\0', sizeof(linha));
   }
   if(cont<48)
      fprintf(stdout, "  ],\n  \"DefaultTemplateData\":\"{ \\\"name\\\":\\\"%s\\\"}\"\n}\n\n\n", DEFAULT_NAME);
   
   free(str);
   fclose(arqin);

   exit(0);
}


